packages/host/app/services/billing-service.ts

```diff
@@ -23,8 +21,17 @@ interface SubscriptionData {
   stripeCustomerId: string | null;
 }
 
+interface StripeLink {
+  type: string;
+  url: string;
+  creditReloadAmount?: number;
+}
+
 export default class BillingService extends Service {
   @tracked private _subscriptionData: SubscriptionData | null = null;
+  @tracked private _stripeLinks: StripeLink[] | null = null;
+  private _fetchingSubscriptionData: Promise<void> | null = null;
+  private _fetchingStripeLinks: Promise<void> | null = null;
```

Comment:
I am wondering if we could simplify this by using [trackedTask](https://reactive.nullvoxpopuli.com/functions/ember_concurrency.trackedTask.html)? Somehow it doesn't sit right with me we need to keep track of promises for simple fetching of data

Reply:
If I understand correctly, `trackedTask` is a resource and it will be a reactive task which means I can use it in the component template to make the template re-render if the state of task is changed. If my understanding is correct, I don't think that's what I need here, I simply need to ensure we fetched the stripe payment links endpoint before we try to get the free plan payment link or the extra credit payment links and open it in the new window.